### PR TITLE
Start of simulator

### PR DESCRIPTION
**Pull request by rsmith49**
Created Deck and BasePlayerHand classes that can be used for simulating games of blackjack.

Also made edits to basic project structure.

Making a PR just in case you were also going to start on the simulator, so I didn't want to crowd the codebase prematurely.

### CODE REVIEW

File: src/simulator/consts.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+BLACKJACK = 21  # Blackjack
+MAX_SCORE = 21  # Idk, maybe they could be different
```
Comment (dtumer) [inline src/simulator/consts.py:2]: Maybe just call is BUST_SCORE

File: src/simulator/deck.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    def shuffle(self):
+        """
+        Shuffles the cards and resets the current card index
+        :return:
+        """
+        self.card_index = 0
+        self.cards = [
+            val for val in CARD_VALUES
+        ] * self.num_decks
```
Comment (dtumer) [inline src/simulator/deck.py:46]: Would it be easier to make this CARD_VALUES * self.num_decks? CARD_VALUES is already a list

File: src/simulator/deck.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        """
+        self.num_decks = num_decks
+        self.reshuffle_at = int(reshuffle_perc * num_decks * CARDS_IN_DECK)
+
+        if random_seed is not None:
+            np.random.seed(random_seed)
+
+        # Put these here so PyCharm would stop yelling at me
+        self.cards = None
+        self.card_index = None
```
Comment (dtumer) [inline src/simulator/deck.py:34]: When we init you could probably just set the index to 0. Might make more sense than None

File: src/simulator/player.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        old_hand = self.hands[self.curr_hand_ndx]
+        # Figured we may as well throw this validation here
+        if len(old_hand.cards) != 2 or old_hand.cards[0] != old_hand.cards[1]:
+            raise ValueError("Can't split on non-matching hand")
+
+        self.hands[self.curr_hand_ndx] = Hand(
+            old_hand.cards[0],
+            self.deck.next_card()
+        )
+        self.hands.insert(
```
Comment (dtumer) [inline src/simulator/player.py:199]: I think you might just be able to use the "append()" function here

File: src/simulator/deck.py (hunk 2 above)
Comment (rsmith49) [reply src/simulator/deck.py:46]: Yeah idk what I was doing here

File: src/simulator/player.py (hunk 4 above)
Comment (rsmith49) [reply src/simulator/player.py:199]: I thought that too, but we can split on the first hand of a split unfortunately

**Pull request merged**